src/pages/calendar/components/Requisites.tsx

```diff
@@ -0,0 +1,176 @@
+                Completed or currently enrolled in {req.quantity} of:{' '}
+              </ListItem>
+            </UnorderedList>
+          )}
+          {req.grade && (
+            <UnorderedList>
+              <ListItem style={{ marginLeft: `${indentationLevel * 40}px` }}>
+                Earn a minimum grade of {req.grade} in each of:
+              </ListItem>
+            </UnorderedList>
+          )}
+          {reqs.map((r) => DisplayRequirement(r, indentationLevel + 1))}{' '}
+          {/* Ensure to increase the indentation level for nested elements */}
+        </Box>
+      );
+    } else {
+      return (
+        <Box>
+          {req.quantity &&
+            !req.grade &&
+            indentationLevel === 1 && ( // Displays this only when its not a grade requirement, and is at the lowest indentation level
+              <UnorderedList>
+                <ListItem style={{ marginLeft: `${indentationLevel * 40}px` }}>
+                  Complete {req.quantity} of the following:{' '}
+                </ListItem>
+              </UnorderedList>
+            )}
+          {req.quantity &&
+            !req.grade &&
+            indentationLevel !== 1 && ( // Displays this only when its not a grade requirement
+              <UnorderedList>
+                <ListItem style={{ marginLeft: `${indentationLevel * 40}px` }}>Complete {req.quantity} of: </ListItem>
+              </UnorderedList>
+            )}
+          {req.grade && (
+            <UnorderedList>
+              <ListItem style={{ marginLeft: `${indentationLevel * 40}px` }}>
+                Earn a minimum grade of {req.grade} in each of the following:
+              </ListItem>
+            </UnorderedList>
+          )}
+          {reqs.map((r) => DisplayRequirement(r, indentationLevel + 1))}{' '}
+          {/* Ensure to increase the indentation level for nested elements */}
+        </Box>
+      );
+    }
+  } else if ('reqListItems' in req) {
+    const nestedReqs = nestedReq(req);
+    return <Box>{nestedReqs}</Box>; // Displays the list of requisites
+  } else if ('code' in req) {
+    // Get the current term and requisite details
+    const subject = req.subject;
+    const code = req.code;
+
+    // Get course details from the backend
+    return (
+      <GetCourseDetails term={currTerm as Term} subject={req.subject} code={req.code}>
+        {(courseDetails) => {
+          if (courseDetails) {
+            // Extract course details and format credits
+            const pid = courseDetails.pid;
+            const credits = courseDetails.credits.credits;
+            const creditsVisual =
+              credits.min === credits.max ? `(${credits.max})` : `(${credits.min} - ${credits.max})`;
+
+            // Render course details with a hyperlink to requisite course page on courseup
+            return (
+              <UnorderedList>
+                <ListItem title="course req" style={{ marginLeft: `${indentationLevel * 40}px` }}>
```

Comment:
This is what shows up when you hover the link so I'd make it more something like `CSC 225 - Algorithms and Data Structures I` or somethin